roles/remediate/tasks/leapp_legacy_network_configuration.yml

```diff
@@ -0,0 +1,70 @@
+---
+- name: leapp_legacy_network_configuration | Convert network manager configuration to the native keyfile format
+  vars:
+    leapp_report_location: /var/log/leapp/leapp-report.json
+    leapp_inhibitor_title: Legacy network configuration found
+  when: 
+    - ansible_distribution == 'RedHat'
+    - ansible_distribution_major_version|int == 9
+  block:
+    - name: leapp_legacy_network_configuration | Check that the leapp-report.json exists
+      ansible.builtin.stat:
+        path: "{{ leapp_report_location }}"
+      register: leapp_report_stat
+
+    - name: leapp_legacy_network_configuration | End remediation early if the leapp report does not exist
+      ansible.builtin.set_fact:
+        leapp_report_missing: true
+      failed_when: true
+      when: 
+        - not leapp_report_stat.stat.exists
+        - leapp_report_stat.stat.size <= 0
+
+    - name: leapp_legacy_network_configuration | Read leapp report
+      ansible.builtin.slurp:
+        src: "{{ leapp_report_location }}"
+      register: leapp_report
+
+    - name: leapp_legacy_network_configuration | Find the inhibitor entry in the leapp report and extract the remediation command
+      # The remediation executes recommended command from the report since it
+      # is generated with specific network interface that needs to be migrated
+      # to the keyfile format.
+      vars:
+        leapp_report_data: "{{ leapp_report.content | b64decode | from_json }}"
+      ansible.builtin.set_fact:
+        leapp_inhibitor_remediation: "{{ item.detail.remediations | selectattr('type', 'eq', 'command') | first }}"
+      loop: "{{ leapp_report_data.entries }}"
+      when: 
+        - item.title == leapp_inhibitor_title
+        - item.detail.remediations | selectattr('type', 'eq', 'command') | list | length > 0
+
+    - name: leapp_legacy_network_configuration | End remediation if a matching entry was not found in the leapp report
+      ansible.builtin.set_fact:
+        leapp_report_missing: true
+      failed_when: true
+      when:
+        - leapp_inhibitor_remediation is not defined
+        - leapp_inhibitor_remediation.context | length <= 0
+
+    - name: leapp_legacy_network_configuration | Show the remediation command
+      ansible.builtin.debug:
+        msg: "{{ leapp_inhibitor_remediation.context | join(' ') }}"
+
+    - name: leapp_legacy_network_configuration | Migrate the network connection configuration to the keyfile format
+      ansible.builtin.command: "{{ leapp_inhibitor_remediation.context | join(' ') }}"
```

Comment:
if `context` is a list with command and arguments like `["keycmd", "-i", "/path/to/something", ....]` then you should use `command` with `argv` instead of converting the list to a space delimited string (which `command` will then have to convert back into a `list`) - https://docs.ansible.com/projects/ansible/latest/collections/ansible/builtin/command_module.html#parameter-argv

```yaml
    - name: leapp_legacy_network_configuration | Migrate the network connection configuration to the keyfile format
      ansible.builtin.command:
        argv: "{{ leapp_inhibitor_remediation.context }}"
```

this is much safer if `context` can contain quotes or other shell metacharacters